mailet/standard/src/test/java/org/apache/james/transport/matchers/TooMuchLinesTest.java

```diff
@@ -0,0 +1,132 @@
+
+package org.apache.james.transport.matchers;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Collection;
+
+import javax.mail.MessagingException;
+
+import org.apache.mailet.MailAddress;
+import org.apache.mailet.base.test.FakeMail;
+import org.apache.mailet.base.test.FakeMatcherConfig;
+import org.apache.mailet.base.test.MimeMessageBuilder;
+import org.junit.Before;
+import org.junit.Test;
+
+public class TooMuchLinesTest {
+
+    private TooMuchLines testee;
+
+    @Before
+    public void setUp() {
+        testee = new TooMuchLines();
+    }
+
+    @Test(expected = MessagingException.class)
+    public void initShouldThrowOnAbsentCondition() throws Exception {
+        testee.init(FakeMatcherConfig.builder().matcherName("name").build());
+    }
+
+    @Test(expected = NumberFormatException.class)
+    public void initShouldThrowOnInvalidCondition() throws Exception {
+        testee.init(
+            FakeMatcherConfig.builder()
+                .condition("a")
+                .matcherName("name")
+                .build());
+    }
+
+    @Test(expected = NumberFormatException.class)
+    public void initShouldThrowOnEmptyCondition() throws Exception {
+        testee.init(FakeMatcherConfig.builder()
+            .condition("")
+            .matcherName("name")
+            .build());
+    }
+
+    @Test(expected = MessagingException.class)
+    public void initShouldThrowOnZeroCondition() throws Exception {
+        testee.init(FakeMatcherConfig.builder()
+            .condition("0")
+            .matcherName("name")
+            .build());
+    }
+
+    @Test(expected = MessagingException.class)
+    public void initShouldThrowOnNegativeCondition() throws MessagingException {
+        testee.init(FakeMatcherConfig.builder()
+            .condition("-10")
+            .matcherName("name")
+            .build());
+    }
+
+    @Test
+    public void matchShouldReturnNoRecipientWhenMailHaveNoMimeMessageAndConditionIs100() throws Exception {
+        testee.init(FakeMatcherConfig.builder()
+            .condition("100")
+            .matcherName("name")
+            .build());
+
+        Collection<MailAddress> result = testee.match(FakeMail.builder().build());
+
+        assertThat(result).isEmpty();
+
+    }
+
+    @Test
+    public void matchShouldAcceptMailsUnderLimit() throws Exception {
+        testee.init(FakeMatcherConfig.builder()
+            .condition("100")
+            .matcherName("name")
+            .build());
+
+        FakeMail fakeMail = FakeMail.builder()
+            .mimeMessage(MimeMessageBuilder.mimeMessageBuilder()
+                .setMultipartWithBodyParts(MimeMessageBuilder.bodyPartBuilder()
+                    .data("content")
+                    .build())
+                .build())
+            .build();
+
+        Collection<MailAddress> result = testee.match(fakeMail);
+
+        assertThat(result).isEmpty();
```

Comment:
The mail don't have too much lines, then we should not match ;-). So we return empty